client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -114,19 +79,56 @@ class EditorMediaModalDetailItem extends Component {
 		return true;
 	}
 
+	/**
+	 * This function return true if the image editor can be
+	 * enabled/shown
+	 *
+	 * @param  {object} item - media item
+	 * @param  {object} site - current site
+	 * @return {boolean} `true` if the image-editor can be enabled.
+	 */
+	enableImageEditing( item, site ) {
+		const { isPrivate } = this.props;
+
+		// do not allow if, for some reason, there isn't a valid item yet
+		if ( ! item ) {
+			return false;
+		}
+
+		// do not show if the feature flag isn't set
+		if ( ! config.isEnabled( 'post-editor/image-editor' ) ) {
+			return false;
+		}
+
+		// do not allow for private sites
+		if ( isPrivate ) {
+			return false;
+		}
+
+		// do not allow for Jetpack site with a non-valid version
+		if (
+			get( site, 'jetpack', false ) &&
+			versionCompare( get( site, 'options.jetpack_version', '0.0' ), '4.7-alpha', '<' )
+		) {
+			return false;
+		}
+
+		return true;
+	}
+
 	renderEditButton() {
 		const { item, onEdit, translate, canUserUploadFiles } = this.props;
 
 		if ( ! canUserUploadFiles ) {
 			return null;
 		}
 
-		if ( this.props.isPrivateSite ) {
+		const mimePrefix = getMimePrefix( item );
+
+		if ( 'image' === mimePrefix && this.props.isPrivateSite ) {
```

Comment:
I may be mistaken here, but it seems that the `isPrivateSite` prop was renamed to `isPrivate` in on line 329.
Is `isPrivateSite` coming from elsewhere? If not we would always be skipping this clause since `if ( 'image' === mimePrefix && this.props.isPrivateSite )` would always be `false`.

Reply:
Great spot! Yes it was and I didn't update this. I think everything was still working as the check happens in two places. 

I've corrected it, but as there are two checks, maybe one should be removed?

Reply:
Ahh I see, because `renderImageEditorButtons` is calling `enableImageEditing` first, which checks for `isSitePrivate` (come to think of it, `enableImageEditing` should probably be renamed to `isImageEditingEnabled` so that it suggests a boolean, vs a side-effect)

In that case I think removing the check from `renderEditButton` would make some sense :)